source/agora/node/Ledger.d

```diff
@@ -1352,3 +1353,88 @@ unittest
     genBlockTransactionsFreeze(1, TxType.Payment);
     assert(ledger.getBlockHeight() == 11 + 2016);
 }
+
+/// test enrollments in the genesis block
+unittest
+{
+    import agora.common.BitField;
+    import std.exception;
+
+    class PreLoadedMemBlockStorage : MemBlockStorage
+    {
+        immutable(Block)[] blocks;
+
+        public this (immutable(Block)[] blocks)
+        {
+            this.blocks = blocks;
+        }
+
+        public override bool load ()
+        {
+            foreach (const ref block; this.blocks)
+                this.saveBlock(block);
+
+            return true;
+        }
+    }
+
+    // 1 payment tx, the rest are freeze txs
+    auto gen_addr = getGenesisKeyPair().address;
+    Transaction FirstTx =
+    {
+        TxType.Payment,
+        inputs: [ Input.init ],
+        outputs: [Output(Amount(62_500_000L * 10_000_000L), gen_addr)],
+    };
+
+    Transaction[] txs;
+    txs ~= FirstTx;
+    Enrollment[] enrolls;
+
+    auto key_pair = KeyPair.random();
+    Transaction tx =
+    {
+        type : TxType.Freeze,
+        inputs : [ Input.init ],
+        outputs : [Output(Amount.MinFreezeAmount, key_pair.address)]
+    };
+
+    txs ~= tx;
+    Hash txhash = hashFull(tx);
+    Hash utxo = UTXOSet.getHash(txhash, 0);
```

Comment:
Is it intentional that there are only 2 transactions ?

Note that we recently had a similar problem with Jay, `makeChainedTransactions` doesn't allow to create frozen tx. That's when I started to poke around the test suite. What I'd do, if we wanted 8 txs, would be to go with `makeChainedTxs` but then change the `type` and re-sign the txs (should be a good 5 lines).
Of course after writing this I realize it doesn't apply because you need to use `Input.init`...

Reply:
> Is it intentional that there are only 2 transactions ?

Yes. #800

Actually I just added a comment about it after I realized it was implicit here.

Reply:
And also yes we need to refactor the transaction creation and make it much more flexible in the future.

Reply:
>     // 1 payment tx, the rest are freeze txs

This comment kinda threw me off, I expect "the rest" to be more than one (other it's "the other").